apps/astarte_appengine_api/lib/astarte_appengine_api_web/plug/health_plug.ex

```diff
@@ -20,65 +20,50 @@ defmodule Astarte.AppEngine.APIWeb.HealthPlug do
   @behaviour Plug
   import Plug.Conn
 
-  alias Astarte.DataAccess.Health.Health
+  alias Astarte.AppEngine.API.Health
 
   def init(_opts) do
     nil
   end
 
   def call(%{request_path: "/health", method: "GET"} = conn, _opts) do
     try do
-      status =
-        case Health.get_health() do
-          {:ok, %{status: :ready}} ->
-            :telemetry.execute(
-              [:astarte, :appengine, :service],
-              %{health: 1},
-              %{consistency_level: :quorum}
-            )
+      status = Health.get_health()
 
-            :telemetry.execute(
-              [:astarte, :appengine, :service],
-              %{health: 1},
-              %{consistency_level: :one}
-            )
-
-            :ok
-
-          {:ok, %{status: :degraded}} ->
-            :telemetry.execute(
-              [:astarte, :appengine, :service],
-              %{health: 0},
-              %{consistency_level: :quorum}
-            )
-
-            :telemetry.execute(
-              [:astarte, :appengine, :service],
-              %{health: 1},
-              %{consistency_level: :one}
-            )
-
-            :ok
-
-          _ ->
-            :telemetry.execute(
-              [:astarte, :appengine, :service],
-              %{health: 0},
-              %{consistency_level: :quorum}
-            )
-
-            :telemetry.execute(
-              [:astarte, :appengine, :service],
-              %{health: 0},
-              %{consistency_level: :one}
-            )
+      with true <- Enum.member?([:ready, :degraded], status) do
+        :telemetry.execute(
+          [:astarte, :appengine, :service],
+          %{health: 1},
+          %{consistency_level: :quorum}
+        )
 
-            :service_unavailable
-        end
+        :telemetry.execute(
+          [:astarte, :appengine, :service],
+          %{health: 1},
+          %{consistency_level: :one}
+        )
 
-      conn
-      |> send_resp(status, "")
-      |> halt()
+        conn
+        |> send_resp(:ok, "")
+        |> halt()
+      else
+        _ ->
+          :telemetry.execute(
+            [:astarte, :appengine, :service],
+            %{health: 0},
+            %{consistency_level: :quorum}
+          )
+
+          :telemetry.execute(
+            [:astarte, :appengine, :service],
+            %{health: 0},
+            %{consistency_level: :one}
+          )
+
+          conn
+          |> send_resp(:service_unavailable, "")
+          |> halt()
+      end
```

Comment:
nit: you're re-wirting things

```suggestion
      {health, status} =
        case Health.get_health() do
          :ready -> {1, :ok}
          :degraded -> {1, :ok}
          :bad -> {0, :service_unavailable}
          :error -> {0, :service_unavailable}
        end
      
      :telemetry.execute(
        [:astarte, :appengine, :service],
        %{health: health},
        %{consistency_level: :quorum}
      )
      
      :telemetry.execute(
        [:astarte, :appengine, :service],
        %{health: health},
        %{consistency_level: :one}
      )
      
      conn
      |> send_resp(status, "")
      |> halt()
```